bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/Thing.java

```diff
@@ -136,4 +153,31 @@
      * @return true if thing is linked, false otherwise.
      */
     public boolean isLinked();
+
+    /**
+     * Returns an immutable copy of the {@link Thing} properties.
+     * 
+     * @return an immutable copy of the {@link Thing} properties (not null)
+     */
+    Map<String, String> getProperties();
+
+    /**
+     * Sets the property value for the property identified by the given name.
+     * 
+     * @param name the name of the property to be set (must not be null or empty)
+     * @param value the value of the property
+     * 
+     * @return the previous value associated with the name, or null if there was no mapping for the name or the previous
+     *         value was null
+     */
+    String setProperty(String name, String value);
+
+    /**
+     * Unsets the property value for the property identified by the given name.
+     * 
+     * @param name the name of the property to be unset (must not be null or empty)
+     * 
+     * @return the previous value associated with the name, or null if there was no mapping for the name
+     */
+    String unsetProperty(String name);
```

Comment:
I agree that "unset" is unusual (at least if you are not an OSGi developer ;)). The reason why I voted for "unset" against "remove" was that "remove" somehow indicates a persistent change, which is not the case here. Remove also don´t fit to "set". Then I would expect a "putProperty" and  instead of "setProperty" (according to the Map interface). 

However, for me "setProperty" with null as value to remove a property is also OK. At the end we just need to document it clearly. So lets go for the "setProperty", which removes a property if the value is null.